client/src/App.js

```diff
@@ -0,0 +1,125 @@
+import React, { Component } from 'react';
+import { Input, Row } from 'react-materialize';
+
+// import D3 from './d3';
+import Chart from './Component/Chart';
+import TopNav from './Component/TopNav';
+import BottomNav from './Component/BottomNav';
+import './App.css';
+
+// cannot use route '/data', seems to be a reserved word?
+const callApi = async (route) => {
+  const response = await fetch(route);
+  const body = await response.json();
+  return body;
+};
+
+class App extends Component {
+  // can I rewrite the state to { stateName: '', statePrice: 0 }?
+  constructor(props) {
+    super(props);
+    this.state = {
+      response: [],
+      level: 'State',
+      year: '2017',
+      stateName: ''
+    }
+    this.getData = this.getData.bind(this);
+    this.handleChange = this.handleChange.bind(this);
```

Comment:
I'm curious why you'd doing these binds, I don't think they should be necessary.

Reply:
I thought every time I declare a method in React class component, I need to bind this method in the constructor... So when I call the method elsewhere (maybe onClick button or onChange input), the `this` is still referring to this component but not button or input.